hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueHandler.java

```diff
@@ -989,8 +991,14 @@ public void markContainerForClose(Container container)
       throws IOException {
     container.writeLock();
     try {
+      ContainerProtos.ContainerDataProto.State state =
+          container.getContainerState();
       // Move the container to CLOSING state only if it's OPEN/RECOVERING
-      if (HddsUtils.isOpenToWriteState(container.getContainerState())) {
+      if (HddsUtils.isOpenToWriteState(state)) {
+        if (state == RECOVERING) {
```

Comment:
Why are we removing here?

Reply:
if we mark a recovering container for close  at datanode, it means this replica has been recovered successfully, so it should not be scrubbed by `staleRecoveringContainerScrubbingService`.